### ISSUE TITLE
Handle key re-exchange initialized from server incorrectly

### ISSUE THREAD

**Issue by jason-ni** [jason-ni/thrussh-mod#4]
According to RFC2453: 
```
9. Key Re-Exchange
 Key re-exchange is started by sending an SSH_MSG_KEXINIT packet when
 not already doing a key exchange (as described in Section 7.1). When
 this message is received, a party MUST respond with its own
 SSH_MSG_KEXINIT message, except when the received SSH_MSG_KEXINIT
 already was a reply.
```
However, code bellow cause server close the session connection. Because the client doesn't response a SSH_MSG_KEXINIT when receiving a server initialized SSH_MSG_KEXINIT.

***code from src/client/encrypted.rs***
```
 56                 } else if let Some(exchange) = std::mem::replace(&mut enc.exchange, None) {
 57                     let kexinit = KexInit::received_rekey(
 58                         exchange,
 59                         negotiation::Client::read_kex(buf, &self.common.config.as_ref().preferred)?,
 60                         &enc.session_id,
 61                     );
 62                     debug!("shall we send a KexInit?");
 63                     enc.rekey = Some(Kex::KexDhDone(kexinit.client_parse(
 64                         self.common.config.as_ref(),
 65                         &mut self.common.cipher,
 66                         buf,
 67                         &mut self.common.write_buffer,
 68                     )?));
 69                 }
```

**Comment by jason-ni** [comment]
Workaround this issue by lower the client side key re-exchange threshold to half of RFC2453 suggested. So the client always initializes the key re-exchange. ff9945daf3ff11905ef84bc28099f9a9055dc319

**Comment by jason-ni** [comment]
It's fixed in thrussh 0.30.6.
